src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -372,4 +373,145 @@ public void deleteInventory(MedicalInventory medicalInventory) throws OHServiceE
+		// TODO: to explore the possibility to allow charges and discharges with same referenceNumber
+		String chargeReferenceNumber = referenceNumber + "-charge";
+		String dischargeReferenceNumber = referenceNumber + "-discharge";
+		MovementType chargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(inventory.getChargeType());
+		MovementType dischargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(inventory.getDischargeType());
+		Supplier supplier = supplierManager.getByID(inventory.getSupplier());
+		Ward ward = wardManager.findWard(inventory.getDestination());
+		LocalDateTime now = TimeTools.getNow();
+		// prepare movements
+		List<Movement> chargeMovements = new ArrayList<>();
+		List<Movement> dischargeMovements = new ArrayList<>();
+		for (MedicalInventoryRow medicalInventoryRow : inventoryRowSearchList) {
+			double theoQty = medicalInventoryRow.getTheoreticQty();
+			double realQty = medicalInventoryRow.getRealQty();
+			Double ajustQty = realQty - theoQty;
+			Medical medical = medicalInventoryRow.getMedical();
+			Lot currentLot = medicalInventoryRow.getLot();
+			if (ajustQty > 0) { // charge movement when realQty > theoQty
+				Movement movement = new Movement(medical, chargeType, null, currentLot, now, ajustQty.intValue(), supplier, chargeReferenceNumber);
+				chargeMovements.add(movement);
+			} else if (ajustQty < 0) { // discharge movement when realQty < theoQty
+				Movement movement = new Movement(medical, dischargeType, ward, currentLot, now, -ajustQty.intValue(), null, dischargeReferenceNumber);
+				dischargeMovements.add(movement);
+			} // else ajustQty = 0, continue
+		}
+		// create movements
+		List<Movement> insertedMovements = new ArrayList<>();
+		if (!chargeMovements.isEmpty()) {
+			insertedMovements.addAll(movStockInsertingManager.newMultipleChargingMovements(chargeMovements, chargeReferenceNumber));
+		}
+		if (!dischargeMovements.isEmpty()) {
+			insertedMovements.addAll(movStockInsertingManager.newMultipleDischargingMovements(dischargeMovements, dischargeReferenceNumber));
+		}
+		String status = InventoryStatus.done.toString();
+		inventory.setStatus(status);
+		this.updateMedicalInventory(inventory, false);
+		return insertedMovements;
+	}
+
+	private void checkReference(MedicalInventory medicalInventory) throws OHServiceException {
+		List<OHExceptionMessage> errors = new ArrayList<>();
+		String reference = medicalInventory.getInventoryReference();
+		String chargeReferenceNumber = reference + "-charge";
+		String dischargeReferenceNumber = reference + "-discharge";
+		boolean existWithSuffixCharge = movStockInsertingManager.refNoExists(chargeReferenceNumber);
+		boolean existWithSuffixDischarge = movStockInsertingManager.refNoExists(dischargeReferenceNumber);
+		MedicalInventory inventory = this.getInventoryByReference(reference);
+		if (existWithSuffixCharge || existWithSuffixDischarge || (inventory != null && inventory.getId() != medicalInventory.getId())) {
```

Comment:
```suggestion
		if (existWithSuffixCharge || existWithSuffixDischarge || inventory != null && inventory.getId() != medicalInventory.getId()) {
```
Unnecessary parentheses